ironfish/src/memPool/feeEstimator.ts

```diff
@@ -152,8 +179,13 @@ export class FeeEstimator {
     }
 
     fees.sort((a, b) => (a < b ? -1 : a > b ? 1 : 0))
+    const lastBlockSize = await this.getLastBlockSizeBytes()
+    const blockSizeRatio = lastBlockSize / this.wallet.chain.consensus.MAX_BLOCK_SIZE_BYTES
+
+    let feeRate = fees[Math.round((queue.length - 1) / 2)]
+    feeRate = BigInt(Math.ceil(Number(feeRate) * blockSizeRatio))
```

Comment:
This won't change the fee rate -- `blockSizeRatio` is between 0 and 1 from the above calculation, so `ceil` will make this equal to `feeRate`.

If we can I think it would be good to avoid converting between number and bigint

Reply:
yeah I really dont like the conversion between bigint and number. But the fee rate is in bigint. But the ratio is a double presented in number, so the ratio can't be in bigint. Is. there any way I can avoid the conversion. I can multiple 100 with ratio to make it as a integer, and divide the final result by 100 and convert back to bigint. What do you think?

Reply:
That sounds like a good approach! We do something a bit like that when estimating the hash rate for the mining pool: https://github.com/iron-fish/ironfish/blob/915e2ee30a823e848bb3cb5801e6194096336bfc/ironfish/src/mining/pool.ts#L411